### PR TITLE
ci: notarize Darwin released binaries

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
Signs darwin binaries

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by dkoshkin** [comment]
Is there any way to test this?

**Comment by supershal** [comment]
> Is there any way to test this in CI?
strategy/hack I used when testing this:
change create a new test workflow file on your branch with following
```
"on":
  push:
    branches:
      - YOUR_BRANCH
```
- instead of pushing the release in last step, use "upload" github action to upload the final artifacts.
Then download artifacts and manually verify them. :)

File: .github/workflows/release-main.yml [hunk 1]
```diff
@@ -90,3 +90,70 @@ jobs:
...
+        name: Run Gon to notarize
+        env:
+          GITHUB_TOKEN: ${{ secrets.MESOSPHERECI_USER_TOKEN }}
+          AC_USERNAME: ${{ secrets.D2IQ_APPLE_DEVELOPER_USERNAME }}
+          AC_PASSWORD: ${{ secrets.D2IQ_APPLE_DEVELOPER_PASSWORD }}
+        run: gon ./gon.hcl -log-level=info -log-json ./config.hcl
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Tar signed bundle
+        run: tar czf konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz konvoy-image-bundle_darwin_amd64
```
Comment (supershal) [inline .github/workflows/release-main.yml:149]: you will have to regenerate checksum file and upload it. I think @alejandroEsc did wrote that script. cant find it.

File: .github/workflows/release-main.yml [hunk 2]
```diff
@@ -90,3 +90,70 @@ jobs:
...
+
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Fetch current main directory
+        uses: actions/checkout@v3
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - if: ${{ needs.release-please.outputs.release_created }}
```
Comment (supershal) [inline .github/workflows/release-main.yml:124]: Should cleanup the certs in the end.
https://github.com/Apple-Actions/import-codesign-certs/issues/8#issuecomment-1011294106

Comment (faiq) [reply .github/workflows/release-main.yml:124]: Ah, yes I commented that earlier.

File: .github/workflows/release-main.yml [hunk 3]
```diff
@@ -161,15 +162,15 @@ jobs:
...
       - name: Calculate checksum
         if: ${{ needs.release-please.outputs.release_created }}
         run: |
-          cat *checksums.txt | grep darwin | xargs -I{} sed -i'' s/{}//g *checksums.txt
-          sed -i'' '/^[[:space:]]*$/d'*checksums.txt
-          sha256sum konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz >> *checksums.txt
-
+          cat *checksums.txt | grep darwin | xargs -I{} sed -i'.bak' s/{}//g *checksums.txt
+          sed -i'.bak' '/^[[:space:]]*$/d' *checksums.txt
+          sha256sum konvoy-image-bundle-v1.19.0_darwin_amd64.tar.gz >> *checksums.txt
```
Comment (faiq) [inline .github/workflows/release-main.yml:173]: oops

File: .github/workflows/release-main.yml [hunk 4]
```diff
@@ -161,15 +162,15 @@ jobs:
         uses: dsaltares/fetch-gh-release-asset@1.0.0
         with:
           version: "tags/${{ needs.release-plese.outputs.release_tag_name }}"
```
Comment (faiq) [inline .github/workflows/release-main.yml:164]: typo

File: .github/workflows/release-main.yml [hunk 5]
```diff
@@ -90,3 +90,94 @@ jobs:
...
+        uses: dsaltares/fetch-gh-release-asset@1.0.0
+        with:
+          version: "tags/${{ needs.release-please.outputs.release_tag_name }}"
+          file: "checksums"
+          regex: true
+
+      - name: Calculate checksum
+        if: ${{ needs.release-please.outputs.release_created }}
+        run: |
+          cat *checksums.txt | grep darwin | xargs -I{} sed -i'.bak' s/{}//g *checksums.txt
```
Comment (supershal) [inline .github/workflows/release-main.yml:171]: awesome

**Review by supershal — APPROVED** [review]

**Pull request merged**